label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleConfigServiceImpl.java

```diff
@@ -102,10 +103,17 @@ public void getLabelApplicationRules(
               .filter(rule -> !labelApplicationRuleIds.contains(rule.getId()))
               .filter(rule -> !deletedSystemLabelApplicationRuleIds.contains(rule.getId()))
               .collect(Collectors.toUnmodifiableList());
+      List<LabelApplicationRule> allLabelApplicationRules =
+          Stream.concat(
+                  labelApplicationRules.stream(), filteredSystemLabelApplicationRules.stream())
+              .filter(
```

Comment:
Please make `LabelApplicationRuleStore` a `IdentifiedObjectStoreWithFilter` and apply the filter there rather than in the impl method. We try to keep the impls logic-free (this one obviously falls a bit short of that goal, but we don't need to make it worse)